SegmentComparison/Logic/vtkSlicerSegmentComparisonModuleLogic.cxx

```diff
@@ -130,21 +130,10 @@ std::string vtkSlicerSegmentComparisonModuleLogicPrivate::GetInputSegmentsAsPlmV
 
   // Get segment binary labelmaps
   vtkSmartPointer<vtkOrientedImageData> referenceSegmentLabelmap = vtkSmartPointer<vtkOrientedImageData>::New();
-  if ( !vtkSlicerSegmentationsModuleLogic::GetSegmentBinaryLabelmapRepresentation(
-    referenceSegmentationNode, referenceSegmentID, referenceSegmentLabelmap ) )
-  {
-    std::string errorMessage("Failed to get binary labelmap from reference segment: " + std::string(referenceSegmentID));
-    vtkErrorMacro("GetInputSegmentsAsPlmVolumes: " << errorMessage);
-    return errorMessage;
-  }
+  referenceSegmentationNode->GetBinaryLabelmapRepresentation(referenceSegmentID, referenceSegmentLabelmap);
+
   vtkSmartPointer<vtkOrientedImageData> compareSegmentLabelmap = vtkSmartPointer<vtkOrientedImageData>::New();
-  if ( !vtkSlicerSegmentationsModuleLogic::GetSegmentBinaryLabelmapRepresentation(
-    compareSegmentationNode, compareSegmentID, compareSegmentLabelmap ) )
-  {
-    std::string errorMessage("Failed to get binary labelmap from reference segment: " + std::string(compareSegmentID));
-    vtkErrorMacro("GetInputSegmentsAsPlmVolumes: " << errorMessage);
-    return errorMessage;
-  }
+  referenceSegmentationNode->GetBinaryLabelmapRepresentation(compareSegmentID, compareSegmentLabelmap);

```

Comment:
We should probably still have some error check and reporting. Maybe the segment is empty or labelmap representation creation fails (out of memory, ...) and in this case it is nicer if some error message is displayed instead of showing some invalid outputs.

Reply:
OK, added back some error checking. Now requires this PR in Slicer: https://github.com/Slicer/Slicer/pull/5193.